tools/mo/openvino/tools/mo/ops/split.py

```diff
@@ -107,6 +108,43 @@ def infer(node):
         elif op == 'AttributedVariadicSplit':
             PermuteAttrs.create_permute_attrs(node, attrs=[('axis', 'input:0')])
 
+    @staticmethod
+    def reverse_infer(node: Node):
+        if node.in_port(0).data.get_shape() is not None:
+            return
+        axis = node.in_port(1).data.get_value() if node.op == 'VariadicSplit' else node.soft_get('axis', None)
+        assert axis is not None, '{} `axis` is unknown for node {}'.format(node.op, node.soft_get('name', node.id))
+
+        split_lengths = node.in_port(2).data.get_value() if node.op == 'VariadicSplit' else node.soft_get('split_lengths', None)
+        assert split_lengths is not None, '{} `split_lengths` is unknown for node {}'.format(node.op, node.soft_get('name', node.id))
+
+        shapes = []
+        splited_dim_size = 0
+        for i in range(len(split_lengths)):
+            shape = node.out_port(i).data.get_shape()
+            if shape is not None:
+                shapes.append(shape)
+            else:
+                # if at least one output shape is None/undefined we cannot define
+                # exact value of shape along axis
+                splited_dim_size = dynamic_dimension
+                continue
+
+            # if input is splitted into several outputs, e.g.
+            # out_shape_1 = [dyn, 4, 3], out_shape_2 = [7, dyn, 3], axis = 2
+            # to get the original source shape [7, 4, 6]
+            # dimensions of axis = 2 must be summed while
+            # for other dimensions clarify_partial_shape should be called
+            splited_dim_size += shapes[-1][axis]
+            shapes[-1][axis] = dynamic_dimension
+
+        if len(shapes) == 0:
+            return
+
+        precise_pshape = clarify_partial_shape(shapes)
```

Comment:
you can't use `clarify_partial_shape` because in case all output static shapes it returns incorrect result and does not sum up dimension values

Reply:
Discussed personally. If we mask shape[axis] as dynamic then we can `clarify_partial_shape` for Split